pkg/hive/cell/config.go

```diff
@@ -165,3 +166,44 @@ func stringToCIDRHookFunc(from reflect.Type, to reflect.Type, data interface{})
 	}
 	return cidr.ParseCIDR(s)
 }
+
+// stringToSliceHookFunc converts strings to []string by splitting by spaces or commas.
+func stringToSliceHookFunc(from reflect.Type, to reflect.Type, data interface{}) (interface{}, error) {
+	if from.Kind() != reflect.String || to.Kind() != reflect.Slice {
+		return data, nil
+	}
+
+	raw := data.(string)
+	if raw == "" {
+		return []string{}, nil
+	}
+
+	// Split by commas first
+	if ss := strings.Split(raw, ","); len(ss) > 1 {
+		return ss, nil
+	}
+
+	// Splitting by commas didn't result in multiple values
+	// so try spaces next.
+	return strings.Split(raw, " "), nil
+}
+
+// fixupStringSliceHookFunc takes a []string coming from pflag.StringSlice and splits it
+// again if it's only a single element. This unifies the flag parsing behavior with StringSlice
+// values coming from environment or configmap where both spaces or commas can be used to split.
+func fixupStringSliceHookFunc(from reflect.Type, to reflect.Type, data interface{}) (interface{}, error) {
```

Comment:
I wonder whether we should introduce this hook or not. I mean, I agree that it brings consistency, but OTOH it is not a regression wrt previous versions, and in a certain sense it goes in the opposite direction of eventually getting rid of the split by space hack.

Reply:
I guess I'm bothered more by the current inconsistency than the possibility that we won't in the end have a single way of splitting it. Would be interested in hearing more opinions on this.

Reply:
> getting rid of the split by space hack.

We might argue that in an alternate reality all expressions of multi-item groupings they'd use commas, but I'm not sure that space-separated lists are necessary a 'hack' or something we can feasibly remove. From a user perspective it could even be considered a feature, for users who do not automatically think in terms of lists being comma-separated.

The support for both comma vs. space approaches, and a reasonable interpretation of mixed spaces + commas  where you *can* express multiple options including spaces seems like a reasonable tradeoff.